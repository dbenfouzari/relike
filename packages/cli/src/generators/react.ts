import { paramCase, snakeCase } from "change-case";
import fs from "fs";
import path from "path";

import { replace, replaceFileName } from "../utils/template";

/**
 * Generates component index file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateIndexFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateIndexFile = (key: string, componentName: string) => {
  const templateIndexPath = path.resolve(__dirname, "../templates/index/index.ts.txt");

  const templateIndex = fs.readFileSync(templateIndexPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateIndexPath, key, componentName)),
    content: replace(templateIndex, key, componentName, paramCase),
  };
};

/**
 * Generates component style file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateStyleFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateStyleFile = (key: string, componentName: string) => {
  const templateStylePath = path.resolve(__dirname, "../templates/react-style/__name__.module.scss.txt");

  const templateStyle = fs.readFileSync(templateStylePath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateStylePath, key, componentName)),
    content: replace(templateStyle, key, componentName, snakeCase),
  };
};

/**
 * Generates component story file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateStoryFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateStoryFile = (key: string, componentName: string) => {
  const templateStoryPath = path.resolve(__dirname, "../templates/react-story/__name__.stories.tsx.txt");

  const templateStory = fs.readFileSync(templateStoryPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateStoryPath, key, componentName)),
    content: replace(templateStory, key, componentName),
  };
};

/**
 * Generates component test file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateTestFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateTestFile = (key: string, componentName: string) => {
  const templateTestPath = path.resolve(__dirname, "../templates/react-test/__name__.test.tsx.txt");

  const templateTest = fs.readFileSync(templateTestPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateTestPath, key, componentName)),
    content: replace(templateTest, key, componentName),
  };
};

/**
 * Generates component file
 *
 * @param   {string}  key           The key to replace
 * @param   {string}  componentName The component name
 * @param   {boolean} hasStyle      Does the component have style
 * @example
 * generateComponentFile("name", name, flags.style);
 * @returns {any}                   File path and content.
 */
export const generateComponentFile = (key: string, componentName: string, hasStyle: boolean) => {
  const templateComponentPath = hasStyle
    ? path.resolve(__dirname, "../templates/react-component/__name__(if:style).tsx.txt")
    : path.resolve(__dirname, "../templates/react-component/__name__.tsx.txt");

  const templateComponent = fs.readFileSync(templateComponentPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateComponentPath, key, componentName)),
    content: replace(templateComponent, key, componentName),
  };
};
