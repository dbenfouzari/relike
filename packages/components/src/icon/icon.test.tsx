import { render, screen } from "@testing-library/react";

import Icons from "../icons";
import Icon from "./icon";

describe("Components/Icon", () => {
  it("should render successfully", () => {
    render(<Icon icon={Icons.access_time} />);

    expect(screen.getByRole("img")).toMatchSnapshot();
  });

  it("should render successfully with rounded variant", () => {
    render(<Icon icon={Icons.access_time_rounded} />);

    const iconComponent = screen.getByRole("img");

    expect(iconComponent).toMatchSnapshot();
    expect(iconComponent).toHaveClass("material-icons-round");
  });

  it("should render successfully with outlined variant", () => {
    render(<Icon icon={Icons.access_time_outlined} />);

    const iconComponent = screen.getByRole("img");

    expect(iconComponent).toMatchSnapshot();
    expect(iconComponent).toHaveClass("material-icons-outlined");
  });

  it("should render successfully with sharp variant", () => {
    render(<Icon icon={Icons.access_time_sharp} />);

    const iconComponent = screen.getByRole("img");

    expect(iconComponent).toMatchSnapshot();
    expect(iconComponent).toHaveClass("material-icons-sharp");
  });

  it("should render successfully with no variant", () => {
    render(<Icon icon={Icons.access_time} />);

    const iconComponent = screen.getByRole("img");

    expect(iconComponent).toMatchSnapshot();
    expect(iconComponent).toHaveClass("material-icons");
  });
});
