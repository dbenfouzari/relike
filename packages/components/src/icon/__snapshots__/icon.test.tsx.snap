// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Icon should render successfully 1`] = `
<span
  class="jsx-3212838394 icon material-icons"
  role="img"
>
  access_time
</span>
`;

exports[`Components/Icon should render successfully with no variant 1`] = `
<span
  class="jsx-3212838394 icon material-icons"
  role="img"
>
  access_time
</span>
`;

exports[`Components/Icon should render successfully with outlined variant 1`] = `
<span
  class="jsx-3212838394 icon material-icons-outlined"
  role="img"
>
  access_time
</span>
`;

exports[`Components/Icon should render successfully with rounded variant 1`] = `
<span
  class="jsx-3212838394 icon material-icons-round"
  role="img"
>
  access_time
</span>
`;

exports[`Components/Icon should render successfully with sharp variant 1`] = `
<span
  class="jsx-3212838394 icon material-icons-sharp"
  role="img"
>
  access_time
</span>
`;
